apps/api/src/app/subscribers/usecases/delete-subscriber-credentials/delete-subscriber-credentials.spec.ts

```diff
@@ -0,0 +1,74 @@
+import { SubscriberRepository } from '@novu/dal';
+import { SubscribersService, UserSession } from '@novu/testing';
+import { Test } from '@nestjs/testing';
+import { expect } from 'chai';
+import { SharedModule } from '../../../shared/shared.module';
+import { ChatProviderIdEnum } from '@novu/shared';
+import { DeleteSubscriberCredentials } from './delete-subscriber-credentials.usecase';
+import { DeleteSubscriberCredentialsCommand } from './delete-subscriber-credentials.command';
+import { UpdateSubscriberChannel } from '../update-subscriber-channel/update-subscriber-channel.usecase';
+import { UpdateSubscriberChannelCommand } from '../update-subscriber-channel/update-subscriber-channel.command';
+import { OAuthHandlerEnum } from '../../types';
+
+describe('Update subscriber provider credentials', function () {
+  let updateSubscriberChannelUsecase: UpdateSubscriberChannel;
+  let deleteSubscriberCredentialsUsecase: DeleteSubscriberCredentials;
+  let session: UserSession;
+  const subscriberRepository = new SubscriberRepository();
+  beforeEach(async () => {
+    const moduleRef = await Test.createTestingModule({
+      imports: [SharedModule],
+      providers: [DeleteSubscriberCredentials, UpdateSubscriberChannel],
+    }).compile();
+
+    session = new UserSession();
+    await session.initialize();
+
+    updateSubscriberChannelUsecase = moduleRef.get<UpdateSubscriberChannel>(UpdateSubscriberChannel);
+    deleteSubscriberCredentialsUsecase = moduleRef.get<DeleteSubscriberCredentials>(DeleteSubscriberCredentials);
+  });
+
+  it('should delete subscriber discord provider credentials', async function () {
+    const subscriberService = new SubscribersService(session.organization._id, session.environment._id);
+    const subscriber = await subscriberService.createSubscriber();
+
+    const subscriberChannel = {
+      providerId: ChatProviderIdEnum.Discord,
+      credentials: { webhookUrl: 'newWebhookUrl' },
+    };
+    await updateSubscriberChannelUsecase.execute(
+      UpdateSubscriberChannelCommand.create({
+        organizationId: subscriber._organizationId,
+        subscriberId: subscriber.subscriberId,
+        environmentId: session.environment._id,
+        providerId: subscriberChannel.providerId,
+        credentials: subscriberChannel.credentials,
+        oauthHandler: OAuthHandlerEnum.NOVU,
+      })
+    );
```

Comment:
I would say that we have to add any other additional channel credentials, and at the end of the test check if it exists... like this way we will be sure that the others wouldn't be deleted

Reply:
I do agree. A second test for that would be very helpful.